sys/kern/kmem.c

```diff
@@ -30,26 +30,29 @@ static void kick_swapper(void) {
 vaddr_t kva_alloc(size_t size) {
   assert(page_aligned_p(size));
   vmem_addr_t start;
+  vaddr_t old;
 
-  bool restarted = false;
-
-retry:
-  if (vmem_alloc(kvspace, size, &start, M_NOGROW)) {
-    if (restarted)
-      return 0;
+  WITH_MTX_LOCK (&maxkvaddr_lock)
```

Comment:
Taking the lock is not necessary if we assume that word-sized memory reads and writes are atomic. FreeBSD makes that assumption, so I think we could make it too.

Reply:
This would be detected by the KCSAN as a data race, because there is a possibility of two accesses to that variable and one of them is write.

Reply:
done